comp/core/remoteagentregistry/impl/remoteagentregistry.go

```diff
@@ -496,23 +609,83 @@ func newRemoteAgentClient(registration *remoteagentregistry.RegistrationData) (p
 	return pb.NewRemoteAgentClient(conn), nil
 }
 
-type remoteAgentDetails struct {
-	lastSeen    time.Time
-	displayName string
-	apiEndpoint string
-	client      pb.RemoteAgentClient
+type configStream struct {
+	ctxCancel     context.CancelFunc
+	configUpdates chan *pb.ConfigUpdate
 }
 
-func newRemoteAgentDetails(registration *remoteagentregistry.RegistrationData) (*remoteAgentDetails, error) {
-	client, err := newRemoteAgentClient(registration)
-	if err != nil {
-		return nil, err
+func (cs *configStream) Cancel() {
+	cs.ctxCancel()
+}
+
+func (cs *configStream) TrySendUpdate(update *pb.ConfigUpdate) bool {
+	select {
+	case cs.configUpdates <- update:
+		return true
+	default:
+		return false
 	}
+}
```

Comment:
I'm wondering if we should put some timer here in case the `configUpdates` channel is full, instead of dropping the update right away. I say this because if we drop the update, it would cause the `runConfigStream` loop to recalculate the snapshot. 
```go
if currentSequenceID > lastEventSequenceID+1 {
	continue outer
}
```

Reply:
I've added a timer of `250 ms`. Open to suggestions for a more appropriate value but, something to consider is that the `agentMap` lock is held during this time.